clients/src/main/java/org/apache/kafka/common/utils/logger/LocationAwareKafkaLogger.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.utils.logger;
+
+import org.slf4j.Marker;
+import org.slf4j.helpers.MessageFormatter;
+import org.slf4j.spi.LocationAwareLogger;
+
+class LocationAwareKafkaLogger extends AbstractKafkaLogger {
```

Comment:
I'm guessing the reason you broke this into separate classes was because you felt `LogContext` was too big with both implementations? I slightly preferred keeping the classes encapsulated since there should never be any reason for another class to access them directly.

Reply:
@hachikuji thanks for the comment. The size was exactly the reason behind this. What I will do is move all these classes inside LogContext.java. Also one question: I have created class 'AbstractKafkaLogger' (as you can see above), one reason for this class was that it helped  in giving consistent name to logger implementations, all the logger implementations ends with 'KafkaLogger'. What is your opinion on this ? Should I keep this class or throw it away ?

Reply:
I think `AbstractKafkaLogger` is a good idea. I'd probably stuff it into `LogContext` as well.